Hate/components/data_transformation.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+import sys
+import string
+import pandas as pd
+import nltk
+from nltk.corpus import stopwords
+nltk.download('stopwords')
+from sklearn.model_selection import train_test_split
+from Hate.logger import logging 
+from Hate.exception import CustomException
+from Hate.entity.config_entity import DataTransformationConfig
+from Hate.entity.artifact_entity import DataIngestionArtifacts, DataTransformationArtifacts
+
+
+class DataTransformation:
+    def __init__(self,data_transformation_config: DataTransformationConfig,data_ingestion_artifacts:DataIngestionArtifacts):
+        self.data_transformation_config = data_transformation_config
+        self.data_ingestion_artifacts = data_ingestion_artifacts
+        
+        
+        
```

Comment:
Add descriptions for the methods calling out their functionalities.

Reply:
Appropriate changes have been made.